File: Task_1/Exponential/exp_taylor.py
def exp_taylor(x, terms ):
    result = 1.0  
    numerator = 1.0  # x^i
    denominator = 1.0  # i!

    for i in range(1, terms):
        numerator *= x  # for x^i
        denominator *= i  #its for i!
        result += numerator / denominator

    return result


# ----------- Main Program -----------

try:
    x = float(input("Enter the value of x for e^x: "))
    
  
    terms_input = input("Enter number of terms in the series (default is 20): ")
    terms = int(terms_input) if terms_input else 20

    if x < 0:
        approx = 1 / exp_taylor((-1)*x, terms)
    else:
        approx = exp_taylor(x, terms)

    print(f"exp({x}) ~ {approx}")

except ValueError:
    print("Invalid input! Please enter numeric values only.")

File: Task_1/Exponential/exp_taylor.c
#include <stdio.h>


double exp_taylor(double x, int terms) {
    double result = 1.0;     
    double numerator = 1.0;  // x^i
    double denominator = 1.0; // i!

    for (int i = 1; i < terms; i++) {
        numerator *= x;       // x^i
        denominator *= i;     // i!
        result += numerator / denominator;
    }

    return result;
}


int main() {
    double x;
    int terms;
    double approx;

    printf("Enter the value of x for e^x: ");
    if (scanf("%lf", &x) != 1) {
        printf("Invalid input! Please enter a numeric value for x.\n");
        return 1;  // Exit with error
    }

    printf("Enter number of terms (default is 20, enter 0 to use default): ");
    if (scanf("%d", &terms) != 1 || terms < 0) {
        printf("Invalid input! Please enter a positive integer for terms.\n");
        return 1;
    }

    // If user enters 0, use default 20 terms
    if (terms == 0)
        terms = 20;

    // Handle negative x
    if (x < 0) {
        approx = 1.0 / exp_taylor(-x, terms);
    } else {
        approx = exp_taylor(x, terms);
    }

    printf("exp(%.2lf) ~ %.10lf\n", x, approx);

    return 0;
}

